webapp/components/sideNav.tsx

```diff
@@ -0,0 +1,55 @@
+import styles from '../styles/SideNav.module.scss';
+import Link from 'next/link';
+import { useState } from 'react';
+import { BiLeftArrow, BiRightArrow } from 'react-icons/bi';
+
+interface SideNavProps {
+	width: string;
+}
+
+const SideNav = (props: SideNavProps) => {
+	const [width, setWidth] = useState(props.width);
+	const closeNav = () => {
+		setWidth('10%');
+	};
+
+	const openNav = () => {
+		setWidth(props.width);
+	};
+
+	return (
+		<div className={styles.sideNav} style={{ width: width }}>
+			<div>
+				{width >= props.width ? (
```

Comment:
we actually don't need 2 buttons here, we can just have a condition to do a certain function. So 1 button is enough.